mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt

```diff
@@ -183,6 +184,10 @@ class WidgetListFragment : Fragment(), WidgetAdapter.ItemClickListener {
                     widget.linkedPage?.let { createShortcut(context, it) }
                     return true
                 }
+                CONTEXT_MENU_ID_OPEN_IN_MAPS -> {
+                    getUrlForMaps(widget.item?.state?.asLocation ?: return true).toUri().openInBrowser(context)
```

Comment:
With the suggestion above, this becomes `widget.item?.state?.asLocation?.toMapsUrl()?.toUri()?.openInBrowser(context)`

Still slightly weird with the large number of question marks, but IMHO better than the return in the middle.

Reply:
Imagine doing that in Java xD